modules/appeals_api/app/workers/appeals_api/monthly_stats_report.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class AppealsApi::MonthlyStatsReport
+  include Sidekiq::Worker
+  include Sidekiq::MonitoredWorker
+
+  sidekiq_options retry: 20, unique_for: 3.weeks
+
+  def perform(end_date: Time.zone.now)
```

Comment:
`perform` arguments must be json-compatible since Sidekiq loads the arguments from a storage location which marshalls down to json; ruby objects like `Time` won't convert properly most the time. Even as a default argument, we should make this apparent (`Time.zone.now.iso8601`) - the date should be coming in as an ISO8601 timestamp that you'll need to convert to a Time in the method (see `AppealReceivedJob#date_submitted`).

You'll also need to go into whichever place(s) are calling the job w/ an `end_date` and convert their timestamps into ISO8601 (see HLR's [update_status](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/app/models/appeals_api/higher_level_review.rb#L285))

Additionally, since Sidekiq doesn't support keyword arguments, change this to a regular argument w/ a default value `perform(end_date = Time.zone.now.iso8601)`

Reply:
I've updated this to make sure `perform` isn't using keywords or non-serializable values, but is it ok that I switched from `Time` to `DateTime`? I found the `DateTime` syntax a little more pleasant when making a static value in the tests - are there other reasons to prefer `Time` though? I did some googling and it seems like they have similar functionality these days

Reply:
No issue with using DateTime if it feels more pleasant :) Going by my anecdotal evidence, `Time` is rather rare in the wild so DateTime is "preferred." I'm pretty sure ActiveRecord date columns (e.g. created_at) respond with a DateTime which probably drove its dominance as well.

Reply:
`Time.current` / `Time.zone.now` are generally what I've seen used in rails projects. My recollection is that it handles timezones more easily than DateTime does, but I admittedly haven't looked into it in a while.

Everything I was able to find on which to use generally tended to point to using [Time instead of DateTime](https://ruby-doc.org/stdlib-2.7.6/libdoc/date/rdoc/DateTime.html#class-DateTime-label-When+should+you+use+DateTime+and+when+should+you+use+Time-3F). Not that we have to worry about it overmuch, but DateTime is deprecated in ruby 3. Might as well keep to the trend.